From: moisseev <moiseev@mezonplus.ru>
Date: Thu, 1 Oct 2020 15:01:51 +0300
Applied-Upstream: https://github.com/rspamd/rspamd/commit/1779fc31f6db9f6250d5302bad5b86094c1e18e1
Subject: Avoid error messages on connect

in read-only mode

Reported by: @nwhisper via Telegram group
---
 interface/js/app/rspamd.js | 9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/interface/js/app/rspamd.js b/interface/js/app/rspamd.js
index 475f040..c3356c7 100644
--- a/interface/js/app/rspamd.js
+++ b/interface/js/app/rspamd.js
@@ -258,6 +258,10 @@ function ($, D3pie, visibility, NProgress, stickyTabs, tab_stat, tab_graph, tab_
     }
 
     function displayUI() {
+        // In many browsers local storage can only store string.
+        // So when we store the boolean true or false, it actually stores the strings "true" or "false".
+        ui.read_only = sessionStorage.getItem("read_only") === "true";
+
         ui.query("auth", {
             success: function (neighbours_status) {
                 $("#selSrv").empty();
@@ -270,15 +274,12 @@ function ($, D3pie, visibility, NProgress, stickyTabs, tab_stat, tab_graph, tab_
                         $('#selSrv [value="' + e.name + '"]').prop("disabled", true);
                     }
                 });
-                tab_selectors.displayUI(ui);
+                if (!ui.read_only) tab_selectors.displayUI(ui);
             },
             errorMessage: "Cannot get server status",
             server: "All SERVERS"
         });
 
-        // In many browsers local storage can only store string.
-        // So when we store the boolean true or false, it actually stores the strings "true" or "false".
-        ui.read_only = sessionStorage.getItem("read_only") === "true";
         if (ui.read_only) {
             $(".ro-disable").attr("disabled", true);
             $(".ro-hide").hide();
